odinmain/src/main/kotlin/me/odinmain/events/EventDispatcher.kt

```diff
@@ -1,24 +1,62 @@
 package me.odinmain.events
 
 import kotlinx.coroutines.*
+import me.odinmain.OdinMain.mc
 import me.odinmain.events.impl.*
 import me.odinmain.utils.*
 import me.odinmain.utils.clock.Clock
+import me.odinmain.utils.skyblock.dungeon.DungeonUtils.inDungeons
+import me.odinmain.utils.skyblock.dungeon.DungeonUtils.isSecret
+import net.minecraft.block.state.IBlockState
 import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.entity.item.EntityItem
+import net.minecraft.entity.passive.EntityBat
 import net.minecraft.inventory.ContainerChest
+import net.minecraft.network.play.client.C08PacketPlayerBlockPlacement
 import net.minecraft.network.play.server.S02PacketChat
+import net.minecraft.network.play.server.S29PacketSoundEffect
 import net.minecraft.network.play.server.S32PacketConfirmTransaction
+import net.minecraft.util.BlockPos
 import net.minecraftforge.client.event.GuiOpenEvent
 import net.minecraftforge.client.event.RenderWorldLastEvent
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
 
 object EventDispatcher {
 
+
+    private val drops = listOf(
+        "Health Potion VIII Splash Potion", "Healing Potion 8 Splash Potion", "Healing Potion VIII Splash Potion", "Healing VIII Splash Potion",
+        "Decoy", "Inflatable Jerry", "Spirit Leap", "Trap", "Training Weights", "Defuse Kit", "Dungeon Chest Key", "Treasure Talisman", "Revive Stone",
+    )
+
+    /**
+     * Dispatches [SecretPickupEvent.Item]
+     */
+    @SubscribeEvent
+    fun onRemoveEntity(event: EntityLeaveWorldEvent) {
+        if (!inDungeons) return
+        if (event.entity is EntityItem && event.entity.entityItem.displayName.noControlCodes.containsOneOf(drops, true)) SecretPickupEvent.Item(event.entity).postAndCatch()
+    }
+
     /**
-     * Dispatches [ChatPacketEvent] and [RealServerTick].
+     * Dispatches [SecretPickupEvent.Interact]
+     */
+    @SubscribeEvent
+    fun onPacket(event: PacketSentEvent) {
+        if (!inDungeons) return
+        val packet = event.packet
```

Comment:
use with(event.packet) {} that gives you the packet as _this_ instead of it being a variable, this should be done other places in the mod too but here for now.